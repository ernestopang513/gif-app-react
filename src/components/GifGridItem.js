import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ({title,url,category = 'GIF'})=> {

    if(!title || title.trim() ==''){
        title = `${category.toUpperCase()} GIF`;
    }
    

    
    return (
        <div className = 'card animate__animated animate__rotateIn' >
            <a href = {url} target = '_blank'>
                <img src= {url} alt={title}/>
            </a>
            <p>{title}</p>
            
        </div>
    )
}

GifGridItem.propTypes = {
    category: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired

}

export default GifGridItem
