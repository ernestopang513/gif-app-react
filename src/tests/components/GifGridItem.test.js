import {shallow} from 'enzyme' ;
import React from 'react';
import GifGridItem from '../../components/GifGridItem';



describe('Pruebas en <GifGridItem />', () => {

    const title = 'Un titulo';
    const url = 'https://soyLaComadreja.com'
    
    test('should mostrar el componente correctamente', () => {
        
        const wrapper = shallow(
            <GifGridItem 
                category = 'Goku' 
                title = {title}
                url = {url}
            />)

        expect(wrapper).toMatchSnapshot()
    })
    
})
