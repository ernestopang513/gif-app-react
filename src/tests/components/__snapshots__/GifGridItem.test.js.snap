// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> should mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__rotateIn"
>
  <a
    href="https://soyLaComadreja.com"
    target="_blank"
  >
    <img
      alt="Un titulo"
      src="https://soyLaComadreja.com"
    />
  </a>
  <p>
    Un titulo
  </p>
</div>
`;
